src/serac/numerics/functional/tensor.hpp

```diff
@@ -390,6 +390,13 @@ SERAC_HOST_DEVICE zero& get(zero& x)
   return x;
 }
 
+/** @brief let `zero` be accessed like a tuple */
+template <int i>
+SERAC_HOST_DEVICE const zero& get(const zero& x)
+{
+  return x;
+}
```

Comment:
```suggestion
template <int i>
SERAC_HOST_DEVICE zero get(zero)
{
  return zero{};
}
```

I don't expect this change to actually make an appreciable difference, but it seems like original implementation is saying "you have to return the same `zero` that was passed in", but it doesn't actually matter where the return value comes from. 

They generate slightly different assembly:
https://godbolt.org/z/3GrvobKo1
I think `xor eax, eax` is slightly preferred, but I'm splitting hairs.

Reply:
Seems reasonable to me!